### DISCUSSION TITLE
Cannot bind `ctrl+c` and `ctrl+v` to copy or paste in linux

### DISCUSSION THREAD

**Discussion by rinormaloku** [ghostty-org/ghostty#3504] [Q&A] [answered]
Cannot bind ctrl+c to copy in linux (Ubuntu 24.04).

If I select it drops the selection but doesn't load anything into the clipboard.

Note: I change my interrupt key to `ctrl+x`, to free up `ctrl+c` and it works in terminals such as Terminator.

This is the keyboard input:

![Image](https://github.com/user-attachments/assets/877ae651-b136-4fd3-9f88-061be26b5759)

**Comment by pandalec** [comment]
There's an open discussion [here](https://github.com/ghostty-org/ghostty/discussions/2811), and yes, I need this too :)

**Comment by rinormaloku** [comment]
I read the discussion and it's about letting `ctrl+c` through if nothing was selected, **in my instance copy doesn't work at all.** 

Would it still be covered by that implementation?

**Comment by haydonryan** [comment]
I'm running arch linux + gnome and copy paste works 

by adding the lines to the config file:

```
keybind = ctrl+c=copy_to_clipboard
keybind = ctrl+v=paste_from_clipboard
```
(what I haven't worked out yet is how to send a ctrl-c (sigint) )

**Reply by xircon** [reply]
```
keybind = ctrl+shift+physical:c=text:\x03
```
Found it in another discussion about russian keyboards.  Works here (UK k/b)

https://github.com/ghostty-org/ghostty/discussions/3584

But it plays havoc with emacs in the terminal, so I abandoned and use ctrl+x to copy.

**Comment by mitchellh** [comment]
I can't reproduce this. Copy works fine for me. Is there a specific keyboard layout or anything else odd going on that can help me reproduce this?

**Reply by rinormaloku** [reply]
I just wanted to reproduce this, and it works for me now. -- weird... I didn't open ghostty since the day I reported the issue, so I cannot point to what could have fixed it.

**Reply by rinormaloku** [reply, marked as answer]
I might have found the reason. Configuration changes are not picked up on config reload, and neither are they picked up when closing and reopening ghostty. 

I had to kill the ghostly instance, which left a lingering process, and thereafter start it to, for example, pick up background opacity. Likely the hotkeys not being picked up was the same issue.

**Comment by xircon** [comment]
Was this aimed at me?  Emacs uses ctrl+c a lot, when I press `ctrl+c a` (if I set `keybind = ctrl+c=copy_to_clipboard`) it swallows the ctrl+c, then the next keystroke just types into the file.

I have a Lenovo Legion laptop, UK keyboard and KDE Plasma (EndeavourOS). I do use keyd to modify some keys (including left control):
```
leftcontrol = overload(control,macro(leftmeta+t))
```
If I tap ctrl, it launches ghostty, cannot see if that would affect it, but will test later, but am on my way out for a while.

**Reply by mitchellh** [reply]
If this was responding to me (you didn't thread so I don't know), then no, I was responding to the OP (hence no thread).

**Comment by xircon** [comment]
Sorry :smile:

**Comment by rinormaloku** [comment]
After trying this once again, copying was working fine! 

I cannot determine what could have fixed it -- besides having rebooted my computer a few times since then I have not used ghostty or made any changes :/
